### ISSUE TITLE
Triggering a scene changes only brightness or color, not both

### ISSUE THREAD

**Issue by fjcaetano** [bwp91/homebridge-govee#26]
### What issue do you have? Please be as thorough and explicit as possible.

I use scenes in the Home app to set the color and brightness in multiple Philips HUE bulbs and set a "mood" to the room. It works correctly for the HUE lamps, however the Govee LED strip only changes its brightness or color when I trigger a scene, but not both.

### Details of your setup.
* Do you use Homebridge (with config-ui?) or HOOBS? 

I'm using Homebridge without UI, with CLI only.

* Which version of Homebridge/HOOBS do you have?

Homebridge 1.1.7

* Which version of this plugin (homebridge-govee) do you have? Has the issue started since upgrading from a previous version?

2.4.0

* Which Govee devices do you have that are causing issues? Please include product models if applicable.

LED Strip H6159

### Please paste any relevant logs below.

<details>

```
Dec 30 10:44:07 raspberrypi homebridge[7995]: [30/12/2020, 10:44:07] [Govee] [getDevices()] HTTP Response:
Dec 30 10:44:07 raspberrypi homebridge[7995]: {
Dec 30 10:44:07 raspberrypi homebridge[7995]:   "data": {
Dec 30 10:44:07 raspberrypi homebridge[7995]:     "devices": [
Dec 30 10:44:07 raspberrypi homebridge[7995]:       {
Dec 30 10:44:07 raspberrypi homebridge[7995]:         "device": "***",
Dec 30 10:44:07 raspberrypi homebridge[7995]:         "model": "H6159",
Dec 30 10:44:07 raspberrypi homebridge[7995]:         "deviceName": "Kitchen",
Dec 30 10:44:07 raspberrypi homebridge[7995]:         "controllable": true,
Dec 30 10:44:07 raspberrypi homebridge[7995]:         "retrievable": true,
Dec 30 10:44:07 raspberrypi homebridge[7995]:         "supportCmds": [
Dec 30 10:44:07 raspberrypi homebridge[7995]:           "turn",
Dec 30 10:44:07 raspberrypi homebridge[7995]:           "brightness",
Dec 30 10:44:07 raspberrypi homebridge[7995]:           "color",
Dec 30 10:44:07 raspberrypi homebridge[7995]:           "colorTem"
Dec 30 10:44:07 raspberrypi homebridge[7995]:         ]
Dec 30 10:44:07 raspberrypi homebridge[7995]:       }
Dec 30 10:44:07 raspberrypi homebridge[7995]:     ]
Dec 30 10:44:07 raspberrypi homebridge[7995]:   },
Dec 30 10:44:07 raspberrypi homebridge[7995]:   "message": "Success",
Dec 30 10:44:07 raspberrypi homebridge[7995]:   "code": 200
Dec 30 10:44:07 raspberrypi homebridge[7995]: }
Dec 30 10:44:07 raspberrypi homebridge[7995]: [30/12/2020, 10:44:07] [Govee] [getDevice()] HTTP Response:
Dec 30 10:44:07 raspberrypi homebridge[7995]: {
Dec 30 10:44:07 raspberrypi homebridge[7995]:   "data": {
Dec 30 10:44:07 raspberrypi homebridge[7995]:     "device": "***",
Dec 30 10:44:07 raspberrypi homebridge[7995]:     "model": "H6159",
Dec 30 10:44:07 raspberrypi homebridge[7995]:     "properties": [
Dec 30 10:44:07 raspberrypi homebridge[7995]:       {
Dec 30 10:44:07 raspberrypi homebridge[7995]:         "online": true
Dec 30 10:44:07 raspberrypi homebridge[7995]:       },
Dec 30 10:44:07 raspberrypi homebridge[7995]:       {
Dec 30 10:44:07 raspberrypi homebridge[7995]:         "powerState": "on"
Dec 30 10:44:07 raspberrypi homebridge[7995]:       },
Dec 30 10:44:07 raspberrypi homebridge[7995]:       {
Dec 30 10:44:07 raspberrypi homebridge[7995]:         "brightness": 73
Dec 30 10:44:07 raspberrypi homebridge[7995]:       },
Dec 30 10:44:07 raspberrypi homebridge[7995]:       {
Dec 30 10:44:07 raspberrypi homebridge[7995]:         "colorTemInKelvin": 2700
Dec 30 10:44:07 raspberrypi homebridge[7995]:       }
Dec 30 10:44:07 raspberrypi homebridge[7995]:     ]
Dec 30 10:44:07 raspberrypi homebridge[7995]:   },
Dec 30 10:44:07 raspberrypi homebridge[7995]:   "message": "Success",
Dec 30 10:44:07 raspberrypi homebridge[7995]:   "code": 200
Dec 30 10:44:07 raspberrypi homebridge[7995]: }
Dec 30 10:44:11 raspberrypi homebridge[7995]: [30/12/2020, 10:44:11] [Govee] [updateDevice()] HTTP Request. This message is yellow for clarity:
Dec 30 10:44:11 raspberrypi homebridge[7995]: {
Dec 30 10:44:11 raspberrypi homebridge[7995]:   "device": "***",
Dec 30 10:44:11 raspberrypi homebridge[7995]:   "model": "H6159",
Dec 30 10:44:11 raspberrypi homebridge[7995]:   "cmd": {
Dec 30 10:44:11 raspberrypi homebridge[7995]:     "name": "brightness",
Dec 30 10:44:11 raspberrypi homebridge[7995]:     "value": 4
Dec 30 10:44:11 raspberrypi homebridge[7995]:   }
Dec 30 10:44:11 raspberrypi homebridge[7995]: }
Dec 30 10:44:11 raspberrypi homebridge[7995]: [30/12/2020, 10:44:11] [Govee] [updateDevice()] HTTP Response:
Dec 30 10:44:11 raspberrypi homebridge[7995]: {
Dec 30 10:44:11 raspberrypi homebridge[7995]:   "code": 200,
Dec 30 10:44:11 raspberrypi homebridge[7995]:   "message": "Success",
Dec 30 10:44:11 raspberrypi homebridge[7995]:   "data": {}
Dec 30 10:44:11 raspberrypi homebridge[7995]: }
Dec 30 10:44:11 raspberrypi homebridge[7995]: [30/12/2020, 10:44:11] [Govee] [Kitchen] setting brightness to [4%].
Dec 30 10:44:11 raspberrypi homebridge[7995]: [30/12/2020, 10:44:11] [Govee] [updateDevice()] HTTP Request. This message is yellow for clarity:
Dec 30 10:44:11 raspberrypi homebridge[7995]: {
Dec 30 10:44:11 raspberrypi homebridge[7995]:   "device": "***",
Dec 30 10:44:11 raspberrypi homebridge[7995]:   "model": "H6159",
Dec 30 10:44:11 raspberrypi homebridge[7995]:   "cmd": {
Dec 30 10:44:11 raspberrypi homebridge[7995]:     "name": "color",
Dec 30 10:44:11 raspberrypi homebridge[7995]:     "value": {
Dec 30 10:44:11 raspberrypi homebridge[7995]:       "r": 249,
Dec 30 10:44:11 raspberrypi homebridge[7995]:       "g": 43,
Dec 30 10:44:11 raspberrypi homebridge[7995]:       "b": 6
Dec 30 10:44:11 raspberrypi homebridge[7995]:     }
Dec 30 10:44:11 raspberrypi homebridge[7995]:   }
Dec 30 10:44:11 raspberrypi homebridge[7995]: }
Dec 30 10:44:11 raspberrypi homebridge[7995]: [30/12/2020, 10:44:11] [Govee] [updateDevice()] HTTP Response:
Dec 30 10:44:11 raspberrypi homebridge[7995]: {
Dec 30 10:44:11 raspberrypi homebridge[7995]:   "code": 200,
Dec 30 10:44:11 raspberrypi homebridge[7995]:   "message": "Success",
Dec 30 10:44:11 raspberrypi homebridge[7995]:   "data": {}
Dec 30 10:44:11 raspberrypi homebridge[7995]: }
Dec 30 10:44:11 raspberrypi homebridge[7995]: [30/12/2020, 10:44:11] [Govee] [Kitchen] setting hue to [9°] and sat to [95%].
Dec 30 10:44:16 raspberrypi homebridge[7995]: [30/12/2020, 10:44:16] [Govee] [updateDevice()] HTTP Request. This message is yellow for clarity:
Dec 30 10:44:16 raspberrypi homebridge[7995]: {
Dec 30 10:44:16 raspberrypi homebridge[7995]:   "device": "***",
Dec 30 10:44:16 raspberrypi homebridge[7995]:   "model": "H6159",
Dec 30 10:44:16 raspberrypi homebridge[7995]:   "cmd": {
Dec 30 10:44:16 raspberrypi homebridge[7995]:     "name": "brightness",
Dec 30 10:44:16 raspberrypi homebridge[7995]:     "value": 73
Dec 30 10:44:16 raspberrypi homebridge[7995]:   }
Dec 30 10:44:16 raspberrypi homebridge[7995]: }
Dec 30 10:44:16 raspberrypi homebridge[7995]: [30/12/2020, 10:44:16] [Govee] [updateDevice()] HTTP Response:
Dec 30 10:44:16 raspberrypi homebridge[7995]: {
Dec 30 10:44:16 raspberrypi homebridge[7995]:   "code": 200,
Dec 30 10:44:16 raspberrypi homebridge[7995]:   "message": "Success",
Dec 30 10:44:16 raspberrypi homebridge[7995]:   "data": {}
Dec 30 10:44:16 raspberrypi homebridge[7995]: }
Dec 30 10:44:16 raspberrypi homebridge[7995]: [30/12/2020, 10:44:16] [Govee] [Kitchen] setting brightness to [73%].
Dec 30 10:44:16 raspberrypi homebridge[7995]: [30/12/2020, 10:44:16] [Govee] [updateDevice()] HTTP Request. This message is yellow for clarity:
Dec 30 10:44:16 raspberrypi homebridge[7995]: {
Dec 30 10:44:16 raspberrypi homebridge[7995]:   "device": "***",
Dec 30 10:44:16 raspberrypi homebridge[7995]:   "model": "H6159",
Dec 30 10:44:16 raspberrypi homebridge[7995]:   "cmd": {
Dec 30 10:44:16 raspberrypi homebridge[7995]:     "name": "color",
Dec 30 10:44:16 raspberrypi homebridge[7995]:     "value": {
Dec 30 10:44:16 raspberrypi homebridge[7995]:       "r": 55,
Dec 30 10:44:16 raspberrypi homebridge[7995]:       "g": 0,
Dec 30 10:44:16 raspberrypi homebridge[7995]:       "b": 255
Dec 30 10:44:16 raspberrypi homebridge[7995]:     }
Dec 30 10:44:16 raspberrypi homebridge[7995]:   }
Dec 30 10:44:16 raspberrypi homebridge[7995]: }
Dec 30 10:44:16 raspberrypi homebridge[7995]: [30/12/2020, 10:44:16] [Govee] [updateDevice()] HTTP Response:
Dec 30 10:44:16 raspberrypi homebridge[7995]: {
Dec 30 10:44:16 raspberrypi homebridge[7995]:   "code": 200,
Dec 30 10:44:16 raspberrypi homebridge[7995]:   "message": "Success",
Dec 30 10:44:16 raspberrypi homebridge[7995]:   "data": {}
Dec 30 10:44:16 raspberrypi homebridge[7995]: }
Dec 30 10:44:16 raspberrypi homebridge[7995]: [30/12/2020, 10:44:16] [Govee] [Kitchen] setting hue to [253°] and sat to [100%].
```

</details>

## Additional Notes

It seems to me that my device cannot set both brightness and color at the same time or too quickly. The REST API doesn't provide a command for setting both so it seems that to achieve that, we need to send two requests to the API. A small delay between each call may also be needed, but I'm not 100% sure about that.

I'm more than happy to work on a PR for this given some guidance. I'm not familiar with Homebridge development, so I couldn't find where to add the delay between API calls to test if that would work.

**Comment by bwp91** [comment]
Hi @fjcaetano 

Do you have this issue with the latest version of the plugin, v2.4.8?

This is strange since your logs all seem "successful".

> It seems to me that my device cannot set both brightness and color at the same time or too quickly. The REST API doesn't provide a command for setting both so it seems that to achieve that, we need to send two requests to the API.

Yes you are correct, in fact if you are also turning the light ON from OFF then there are three different commands to be sent

* to turn on
* to set the brightness
* to set the colour

> A small delay between each call may also be needed, but I'm not 100% sure about that.

In the plugin I have dealt with this by adding a small delay offset for when commands are sent:

* on/off commands are sent immediately
* brightness commands are sent after 750ms
* colour commands are sent after 1500ms

Perhaps these delays need to be increased?

Please update to the latest version of the plugin to see if you still experience this issue, and if so we can look into this further.

Thanks

**Comment by fjcaetano** [comment]
I've updated the lib but had the same results:

```
<triggering new scene>

Dec 31 14:07:30 raspberrypi homebridge[30031]: [31/12/2020, 14:07:30] [Govee] [updateDevice()] HTTP Request. This message is yellow for clarity:
Dec 31 14:07:30 raspberrypi homebridge[30031]: {
Dec 31 14:07:30 raspberrypi homebridge[30031]:   "device": "***",
Dec 31 14:07:30 raspberrypi homebridge[30031]:   "model": "H6159",
Dec 31 14:07:30 raspberrypi homebridge[30031]:   "cmd": {
Dec 31 14:07:30 raspberrypi homebridge[30031]:     "name": "turn",
Dec 31 14:07:30 raspberrypi homebridge[30031]:     "value": "on"
Dec 31 14:07:30 raspberrypi homebridge[30031]:   }
Dec 31 14:07:30 raspberrypi homebridge[30031]: }
Dec 31 14:07:30 raspberrypi homebridge[30031]: [31/12/2020, 14:07:30] [Govee] [updateDevice()] HTTP Response:
Dec 31 14:07:30 raspberrypi homebridge[30031]: {
Dec 31 14:07:30 raspberrypi homebridge[30031]:   "code": 200,
Dec 31 14:07:30 raspberrypi homebridge[30031]:   "message": "Success",
Dec 31 14:07:30 raspberrypi homebridge[30031]:   "data": {}
Dec 31 14:07:30 raspberrypi homebridge[30031]: }
Dec 31 14:07:30 raspberrypi homebridge[30031]: [31/12/2020, 14:07:30] [Govee] [Kitchen] setting state to [on].

<triggering new scene>

Dec 31 14:07:40 raspberrypi homebridge[30031]: [31/12/2020, 14:07:40] [Govee] [getDevices()] HTTP Response:
Dec 31 14:07:40 raspberrypi homebridge[30031]: {
Dec 31 14:07:40 raspberrypi homebridge[30031]:   "data": {
Dec 31 14:07:40 raspberrypi homebridge[30031]:     "devices": [
Dec 31 14:07:40 raspberrypi homebridge[30031]:       {
Dec 31 14:07:40 raspberrypi homebridge[30031]:         "device": "***",
Dec 31 14:07:40 raspberrypi homebridge[30031]:         "model": "H6159",
Dec 31 14:07:40 raspberrypi homebridge[30031]:         "deviceName": "Kitchen",
Dec 31 14:07:40 raspberrypi homebridge[30031]:         "controllable": true,
Dec 31 14:07:40 raspberrypi homebridge[30031]:         "retrievable": true,
Dec 31 14:07:40 raspberrypi homebridge[30031]:         "supportCmds": [
Dec 31 14:07:40 raspberrypi homebridge[30031]:           "turn",
Dec 31 14:07:40 raspberrypi homebridge[30031]:           "brightness",
Dec 31 14:07:40 raspberrypi homebridge[30031]:           "color",
Dec 31 14:07:40 raspberrypi homebridge[30031]:           "colorTem"
Dec 31 14:07:40 raspberrypi homebridge[30031]:         ]
Dec 31 14:07:40 raspberrypi homebridge[30031]:       }
Dec 31 14:07:40 raspberrypi homebridge[30031]:     ]
Dec 31 14:07:40 raspberrypi homebridge[30031]:   },
Dec 31 14:07:40 raspberrypi homebridge[30031]:   "message": "Success",
Dec 31 14:07:40 raspberrypi homebridge[30031]:   "code": 200
Dec 31 14:07:40 raspberrypi homebridge[30031]: }
Dec 31 14:07:40 raspberrypi homebridge[30031]: [31/12/2020, 14:07:40] [Govee] [getDevice()] HTTP Response:
Dec 31 14:07:40 raspberrypi homebridge[30031]: {
Dec 31 14:07:40 raspberrypi homebridge[30031]:   "data": {
Dec 31 14:07:40 raspberrypi homebridge[30031]:     "device": "***",
Dec 31 14:07:40 raspberrypi homebridge[30031]:     "model": "H6159",
Dec 31 14:07:40 raspberrypi homebridge[30031]:     "properties": [
Dec 31 14:07:40 raspberrypi homebridge[30031]:       {
Dec 31 14:07:40 raspberrypi homebridge[30031]:         "online": true
Dec 31 14:07:40 raspberrypi homebridge[30031]:       },
Dec 31 14:07:40 raspberrypi homebridge[30031]:       {
Dec 31 14:07:40 raspberrypi homebridge[30031]:         "powerState": "on"
Dec 31 14:07:40 raspberrypi homebridge[30031]:       },
Dec 31 14:07:40 raspberrypi homebridge[30031]:       {
Dec 31 14:07:40 raspberrypi homebridge[30031]:         "brightness": 73
Dec 31 14:07:40 raspberrypi homebridge[30031]:       },
Dec 31 14:07:40 raspberrypi homebridge[30031]:       {
Dec 31 14:07:40 raspberrypi homebridge[30031]:         "color": {
Dec 31 14:07:40 raspberrypi homebridge[30031]:           "r": 255,
Dec 31 14:07:40 raspberrypi homebridge[30031]:           "b": 0,
Dec 31 14:07:40 raspberrypi homebridge[30031]:           "g": 0
Dec 31 14:07:40 raspberrypi homebridge[30031]:         }
Dec 31 14:07:40 raspberrypi homebridge[30031]:       }
Dec 31 14:07:40 raspberrypi homebridge[30031]:     ]
Dec 31 14:07:40 raspberrypi homebridge[30031]:   },
Dec 31 14:07:40 raspberrypi homebridge[30031]:   "message": "Success",
Dec 31 14:07:40 raspberrypi homebridge[30031]:   "code": 200
Dec 31 14:07:40 raspberrypi homebridge[30031]: }
Dec 31 14:07:40 raspberrypi homebridge[30031]: [31/12/2020, 14:07:40] [Govee] [updateDevice()] HTTP Request. This message is yellow for clarity:
Dec 31 14:07:40 raspberrypi homebridge[30031]: {
Dec 31 14:07:40 raspberrypi homebridge[30031]:   "device": "***",
Dec 31 14:07:40 raspberrypi homebridge[30031]:   "model": "H6159",
Dec 31 14:07:40 raspberrypi homebridge[30031]:   "cmd": {
Dec 31 14:07:40 raspberrypi homebridge[30031]:     "name": "brightness",
Dec 31 14:07:40 raspberrypi homebridge[30031]:     "value": 4
Dec 31 14:07:40 raspberrypi homebridge[30031]:   }
Dec 31 14:07:40 raspberrypi homebridge[30031]: }
Dec 31 14:07:41 raspberrypi homebridge[30031]: [31/12/2020, 14:07:41] [Govee] [updateDevice()] HTTP Response:
Dec 31 14:07:41 raspberrypi homebridge[30031]: {
Dec 31 14:07:41 raspberrypi homebridge[30031]:   "code": 200,
Dec 31 14:07:41 raspberrypi homebridge[30031]:   "message": "Success",
Dec 31 14:07:41 raspberrypi homebridge[30031]:   "data": {}
Dec 31 14:07:41 raspberrypi homebridge[30031]: }
Dec 31 14:07:41 raspberrypi homebridge[30031]: [31/12/2020, 14:07:41] [Govee] [Kitchen] setting brightness to [4%].
Dec 31 14:07:41 raspberrypi homebridge[30031]: [31/12/2020, 14:07:41] [Govee] [updateDevice()] HTTP Request. This message is yellow for clarity:
Dec 31 14:07:41 raspberrypi homebridge[30031]: {
Dec 31 14:07:41 raspberrypi homebridge[30031]:   "device": "***",
Dec 31 14:07:41 raspberrypi homebridge[30031]:   "model": "H6159",
Dec 31 14:07:41 raspberrypi homebridge[30031]:   "cmd": {
Dec 31 14:07:41 raspberrypi homebridge[30031]:     "name": "color",
Dec 31 14:07:41 raspberrypi homebridge[30031]:     "value": {
Dec 31 14:07:41 raspberrypi homebridge[30031]:       "r": 249,
Dec 31 14:07:41 raspberrypi homebridge[30031]:       "g": 43,
Dec 31 14:07:41 raspberrypi homebridge[30031]:       "b": 6
Dec 31 14:07:41 raspberrypi homebridge[30031]:     }
Dec 31 14:07:41 raspberrypi homebridge[30031]:   }
Dec 31 14:07:41 raspberrypi homebridge[30031]: }
Dec 31 14:07:41 raspberrypi homebridge[30031]: [31/12/2020, 14:07:41] [Govee] [updateDevice()] HTTP Response:
Dec 31 14:07:41 raspberrypi homebridge[30031]: {
Dec 31 14:07:41 raspberrypi homebridge[30031]:   "code": 200,
Dec 31 14:07:41 raspberrypi homebridge[30031]:   "message": "Success",
Dec 31 14:07:41 raspberrypi homebridge[30031]:   "data": {}
Dec 31 14:07:41 raspberrypi homebridge[30031]: }
Dec 31 14:07:41 raspberrypi homebridge[30031]: [31/12/2020, 14:07:41] [Govee] [Kitchen] setting hue to [9°] and sat to [95%].

<triggering new scene>

Dec 31 14:07:48 raspberrypi homebridge[30031]: [31/12/2020, 14:07:48] [Govee] [updateDevice()] HTTP Request. This message is yellow for clarity:
Dec 31 14:07:48 raspberrypi homebridge[30031]: {
Dec 31 14:07:48 raspberrypi homebridge[30031]:   "device": "***",
Dec 31 14:07:48 raspberrypi homebridge[30031]:   "model": "H6159",
Dec 31 14:07:48 raspberrypi homebridge[30031]:   "cmd": {
Dec 31 14:07:48 raspberrypi homebridge[30031]:     "name": "brightness",
Dec 31 14:07:48 raspberrypi homebridge[30031]:     "value": 73
Dec 31 14:07:48 raspberrypi homebridge[30031]:   }
Dec 31 14:07:48 raspberrypi homebridge[30031]: }
Dec 31 14:07:48 raspberrypi homebridge[30031]: [31/12/2020, 14:07:48] [Govee] [updateDevice()] HTTP Response:
Dec 31 14:07:48 raspberrypi homebridge[30031]: {
Dec 31 14:07:48 raspberrypi homebridge[30031]:   "code": 200,
Dec 31 14:07:48 raspberrypi homebridge[30031]:   "message": "Success",
Dec 31 14:07:48 raspberrypi homebridge[30031]:   "data": {}
Dec 31 14:07:48 raspberrypi homebridge[30031]: }
Dec 31 14:07:48 raspberrypi homebridge[30031]: [31/12/2020, 14:07:48] [Govee] [Kitchen] setting brightness to [73%].
Dec 31 14:07:48 raspberrypi homebridge[30031]: [31/12/2020, 14:07:48] [Govee] [updateDevice()] HTTP Request. This message is yellow for clarity:
Dec 31 14:07:48 raspberrypi homebridge[30031]: {
Dec 31 14:07:48 raspberrypi homebridge[30031]:   "device": "***",
Dec 31 14:07:48 raspberrypi homebridge[30031]:   "model": "H6159",
Dec 31 14:07:48 raspberrypi homebridge[30031]:   "cmd": {
Dec 31 14:07:48 raspberrypi homebridge[30031]:     "name": "color",
Dec 31 14:07:48 raspberrypi homebridge[30031]:     "value": {
Dec 31 14:07:48 raspberrypi homebridge[30031]:       "r": 55,
Dec 31 14:07:48 raspberrypi homebridge[30031]:       "g": 0,
Dec 31 14:07:48 raspberrypi homebridge[30031]:       "b": 255
Dec 31 14:07:48 raspberrypi homebridge[30031]:     }
Dec 31 14:07:48 raspberrypi homebridge[30031]:   }
Dec 31 14:07:48 raspberrypi homebridge[30031]: }
Dec 31 14:07:48 raspberrypi homebridge[30031]: [31/12/2020, 14:07:48] [Govee] [updateDevice()] HTTP Response:
Dec 31 14:07:48 raspberrypi homebridge[30031]: {
Dec 31 14:07:48 raspberrypi homebridge[30031]:   "code": 200,
Dec 31 14:07:48 raspberrypi homebridge[30031]:   "message": "Success",
Dec 31 14:07:48 raspberrypi homebridge[30031]:   "data": {}
Dec 31 14:07:48 raspberrypi homebridge[30031]: }
Dec 31 14:07:48 raspberrypi homebridge[30031]: [31/12/2020, 14:07:48] [Govee] [Kitchen] setting hue to [253°] and sat to [100%].

<triggering new scene>

```

As you can see above, I triggered 3 different scenes and the LED strip never changed the color being used, only the brightness. Triggering the last scene didn't call `updateDevice()`, it just called `getDevice()` as it does automatically after a period of time.

When I triggered the first scene, the LED strip was turned off so the `turn on` command was sent, but not the brightness and color updates to match the scene.

I think the weirdest thing is that the lib is sending commands to update the color, but it does not affect the LED strip. All of these tests were done with the LED with a pure red `#F00` color since none of my scenes has that color, but the strip kept red through all the scenes. I wonder if not sending the brightness change command would fix the color change and/or if reverting the order of the commands would yield any results since it seems the lib always change the brightness before changing the color

**Comment by bwp91** [comment]
The logs you have posted aren't consistent with v2.4.8. Did you restart HB after updating?

**Comment by fjcaetano** [comment]
Hey @bwp91, you're correct, I thought I had updated it but I didn't.

The LED strip's behavior is much closer to the expected now, the only issue is that there were a couple of times nothing happened after updating the device on the Home app either. Setting the brightness always works, but setting colors sometimes just doesn't work. There is no logging or anything. I've tried commenting [this line](https://github.com/bwp91/homebridge-govee/blob/d295aebbdec67c517f5007de528b508381aca2d7/lib/device/led-strip.js#L104) in case it's something to do with the cached values, but the behavior was the same.

I'm not 100% sure of what might be going on, if there's an error happening or if the device is receiving the request from Homebridge at all. Warnings and errors should also show up in the Homebridge logs along with the debug ones, right?

Any idea of what might be happening or how to investigate it further?
